plugins_examples/archive/archive.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+	"regexp"
+)
+
+func main() {
+	s := bufio.NewScanner(os.Stdin) 
+	for s.Scan() {
+		line := s.Text()
+		if strings.Contains(line, "PRIVMSG") == true {
+			skip := false;
+
+	    		re := regexp.MustCompile("https?\\:\\/\\/[[:^space:]]*")
+	    		match := re.FindString(line)
+
+			if match != "" && match != "https://arch.cccp.io/urls.txt" {
+	    			fr, err := os.Open("/var/www/html/arch/urls.txt")
+	    			if err != nil {
+				    fmt.Fprintln(os.Stderr, "error", err)
+	    			}
+				defer fr.Close()
```

Comment:
`defer` adds the `fr.Close()` to the stack of the surrounding function, in this case, `main`:

> A defer statement pushes a function call onto a list. The list of saved calls is executed after the surrounding function returns. Defer is commonly used to simplify functions that perform various clean-up actions. 

You're opening the `fr` file inside a for loop and doing `defer fr.Close()`  in each iteration of the loop. When the loop is finished and plugin stops, a lot of fr.Close will be executed. Meanwhile you have open the same file several time. I suggest you remove `defer` here, and move the `fr.Close()` to the end of the for loop, so the file is closed properly everytime.

Another way: you can open the file Read-Only at the beginning of `main`, outside the loop, and add the `defer fr.Close()` right after, so it is done once. But you will have to find a way to set the position to the beginning of the file for your Scanner to parse the whole file.